pkg/processor/trigger/partitioned/v3io/partition.go

```diff
@@ -57,49 +57,26 @@ func newPartition(parentLogger logger.Logger, v3ioTrigger *v3io, partitionID int
 func (p *partition) Read() error {
 	partitionPath := fmt.Sprintf("%s/%d", p.v3ioTrigger.streamPath, p.partitionID)
 
-	p.Logger.DebugWith("Seeking partition",
-		"partitionPath", partitionPath,
-		"seekType", p.v3ioTrigger.seekType)
-
-	response, err := p.v3ioTrigger.container.Sync.SeekShard(&v3iohttp.SeekShardInput{
-		Path: partitionPath,
-		Type: p.v3ioTrigger.seekType,
-	})
-
+	location, err := p.seek(partitionPath)
 	if err != nil {
-		return errors.Wrap(err, "Failed to seek partition")
-	}
-
-	// get the location
-	location := response.Output.(*v3iohttp.SeekShardOutput).Location
-	if location == "" {
-		return errors.New("Got empty location from seek")
+		return err
 	}
 
 	p.Logger.DebugWith("Starting to read from partition",
 		"location", location,
 		"pollingInterval", p.v3ioTrigger.configuration.PollingIntervalMs)
 
-	// release seek shard response
-	response.Release()
-
 	pollingInterval := time.Duration(p.v3ioTrigger.configuration.PollingIntervalMs) * time.Millisecond
 
 	for {
-		// get records
-		response, err = p.v3ioTrigger.container.Sync.GetRecords(&v3iohttp.GetRecordsInput{
-			Path:     partitionPath,
-			Location: location,
-			Limit:    p.v3ioTrigger.configuration.ReadBatchSize,
-		})
-
-		// TODO: skip errors
+		getRecordsOutput, err := p.getRecords(partitionPath, location, pollingInterval)
+
 		if err != nil {
-			return errors.Wrap(err, "Failed to read from partition")
+			p.Logger.ErrorWith("Failed to read from partition", "error", err)
+			time.Sleep(pollingInterval)
```

Comment:
polling interval is allowed to be zero, please don't use it

Reply:
What's wrong with zero?

Reply:
if GetRecords returns errors immediately for a long period of time, this will cause an infinite loop that will just devour cpu resources with no good purpose.
We don't expect to hit this logic often, but we should be cautious if it will.

Reply:
But that's exactly the purpose of setting zero (the default is 500ms).

Reply:
the purpose is to wait as little as possible between requests that the server returned an error on them, and that can be zero since doing an io operation takes a long time.
on the client side it's best not to go into busywait loop when the error may be local.